File: dotnet-desktop-guide/net/winforms/controls-design/snippets/designer-properties-overview/vb/CompassRose.vb
Imports System.ComponentModel

Public Class CompassRose

    '<defaultvalue>
    <DefaultValue(GetType(Directions), "North")>
    Public Property PointerDirection As Directions = Directions.North

    <DefaultValue(10)>
    Public Property DistanceInFeet As Integer = 10
    '</defaultvalue>

    '<browsable>
    <Browsable(False)>
    Public Property IsSelected As Boolean
    '</browsable>

    '<shouldserialize_reset>
    Public Property Direction As Directions = Directions.None

    Private Sub ResetDirection()
        Direction = Directions.None
    End Sub

    Private Function ShouldSerializeDirection() As Boolean
        Return Direction <> Directions.None
    End Function
    '</shouldserialize_reset>
End Class

File: dotnet-desktop-guide/net/winforms/controls-design/snippets/designer-properties-overview/csharp/CompassRose.cs
using System.ComponentModel;

namespace UserControlProject;

public partial class CompassRose : UserControl
{
    // <defaultvalue>
    [DefaultValue(typeof(Directions), "North")]
    public Directions PointerDirection { get; set; } = Directions.North;

    [DefaultValue(10)]
    public int DistanceInFeet { get; set; } = 10;
    // </defaultvalue>

    // <browsable>
    [Browsable(false)]
    public bool IsSelected { get; set; }
    // </browsable>

    public CompassRose() =>
        InitializeComponent();

    // <shouldserialize_reset>
    public Directions Direction { get; set; } = Directions.None;

    private void ResetDirection() =>
        Direction = Directions.None;

    private bool ShouldSerializeDirection() =>
        Direction != Directions.None;
    // </shouldserialize_reset>
}
